src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -216,7 +216,9 @@ static void uct_rc_mlx5_iface_progress_enable(uct_iface_h tl_iface, unsigned fla
 {
     uct_rc_mlx5_iface_common_t *iface = ucs_derived_of(tl_iface, uct_rc_mlx5_iface_common_t);
 
-    if (flags & UCT_PROGRESS_RECV) {
+    if ((flags & UCT_PROGRESS_RECV) &&
+        (iface->super.rx.srq.quota > 0)) { /* prepost recvs only if quota available
+                                            * (recvs were not preposted before) */
```

Comment:
1. nice catch 👍 
2. is it fixing the srq avail assertion? can you add this to PR description?
3. maybe move the >0 check inside uct_rc_mlx5_iface_common_prepost_recvs?

Reply:
4. pls submit to v1.10.x as well

Reply:
imo, >0 check better to keep here, because `uct_rc_mlx5_iface_common_prepost_recvs` is called much more frequently, so check would add additional overhead

Reply:
i see uct_rc_mlx5_iface_common_prepost_recvs is called only from uct_rc_mlx5_iface_progress_enable and dc iface init, what am i missing?

Reply:
BTW, probably good to add ucs_assert(available>0) to uct_rc_mlx5_iface_srq_post_recv

Reply:
may fault, mixed it with `uct_rc_mlx5_iface_srq_post_recv`

Reply:
> is it fixing the srq avail assertion? can you add this to PR description?

yes, updated PR/commit comment

> maybe move the >0 check inside uct_rc_mlx5_iface_common_prepost_recvs?

moved to ```uct_rc_mlx5_iface_common_prepost_recvs```

> good to add ucs_assert(available>0) to uct_rc_mlx5_iface_srq_post_recv

added new assert

> pls submit to v1.10.x as well

will do backport after merge into master